qa/backwards/shared/src/main/java/org/elasticsearch/test/ExternalNodeService.java

```diff
@@ -373,10 +349,14 @@ private void start(Deque<String> commandLine) throws IOException {
             Process process = null;
             String port = null;
             String pid = null;
-            BufferedReader stdout = null;
             try {
                 process = builder.start();
-                stdout = new BufferedReader(new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8));
+                final PipedInputStream inputFromExternalNode = new PipedInputStream();
+                final PipedOutputStream outputFromExternalNode= new PipedOutputStream(inputFromExternalNode);
+                BufferedReader stdout = new BufferedReader(new InputStreamReader(inputFromExternalNode, StandardCharsets.UTF_8));
+                final AtomicBoolean writeToPipe = new AtomicBoolean(true);
```

Comment:
I remember this change now!

I wonder if you can drop the atomic boolean and the pipes and stuff and use a TeeOutputStream or something. This just feels complicated.

Maybe only start the LoggerThread after the process has started fully? Before that just read from stdout like we were before?